types/src/on_chain_config/aptos_features.rs

```diff
@@ -274,6 +279,29 @@ impl Features {
     pub fn is_refundable_bytes_enabled(&self) -> bool {
         self.is_enabled(FeatureFlag::REFUNDABLE_BYTES)
     }
+
+    pub fn get_max_identifier_size(&self) -> u64 {
+        if self.is_enabled(FeatureFlag::LIMIT_MAX_IDENTIFIER_LENGTH) {
+            IDENTIFIER_SIZE_MAX
+        } else {
+            LEGACY_IDENTIFIER_SIZE_MAX
+        }
+    }
+
+    pub fn get_max_binary_format_version(&self, gas_feature_version_opt: Option<u64>) -> u32 {
+        // For historical reasons, we support still < gas version 5, but if a new caller don't specify
+        // the gas version, we default to 5, which was introduced in late '22.
+        let gas_feature_version = gas_feature_version_opt.unwrap_or(5);
+        if gas_feature_version < 5 {
```

Comment:
HM, is there any invariant that we can check here? Seems like the only thing we need from that parameter is whether it's None or <5? what happens if some large number is provided? Couldn't we just pass a boolean instead?

Reply:
Updated to not take gas feature version into account